gulpfile.js

```diff
@@ -55,6 +65,48 @@ gulp.task('images', function buildImages() {
     .pipe(gulp.dest(paths.images.dest));
 });
 
+/**
+ * Adds necessary AMP script tags.
+ *
+ * Replaces the placeholder '${ampjs}' with the AMP base script and any
+ * necessary scripts to support custom elements.
+ *
+ * @param {!Vinyl} file The file to scan and add tags to.
+ * @return {!Vinyl} The modified file.
+ */
+function addScriptTags(file) {
+  const dom = new JSDOM(file.contents.toString());
+  const doc = dom.window.document;
+  const ampTags = new Set(Array.from(doc.getElementsByTagName('*'))
+      .map(e => e.tagName.toLowerCase())
+      .filter(t => t.startsWith('amp'))
+      .filter(t => !AMP_EXCLUDED_TAGS.has(t))
+      .map(t => AMP_REMAPPED_TAGS[t] || t));
+  const urls = [AMP_BASE_URL_ELEMENT, ...Array.from(ampTags, t => {
+    return `<script async custom-element="${t}" src="https://cdn.ampproject.org/v0/${t}-latest.js"></script>`;
+  })];
+  file.contents = new Buffer(file.contents.toString().replace(AMP_PLACEHOLDER,
+      urls.join('\n')));
+  return file;
+}
+
+/**
+ * Creates the include function for use in .pipe()
+ *
+ * @return {Function} The function for use in gulp.
+ */
+function includeAmpCustomElements() {
```

Comment:
Is there a prettier way to wrap addScriptTags() so that it can be used by gulp?  If not, then carry on :)  But should includeAmpCustomElements() have a more similar name to addScriptTags() at least?